### PR TITLE
Support categories

### PR DESCRIPTION
**Pull request by ilabsvictor**

### CODE REVIEW

File: AffirmSDK/AffirmItem.h [hunk 1]
```diff
@@ -67,8 +72,27 @@ NS_ASSUME_NONNULL_BEGIN
...
+ @param URL URL of the item.
+ @param categories An array of lists that indicate the various categories that apply to this product, and the hierarchy of those category definitions.
+ @return The newly created item.
+ */
++ (AffirmItem *)itemWithName:(NSString *)name
+                         SKU:(NSString *)SKU
+                   unitPrice:(NSDecimalNumber *)unitPrice
+                    quantity:(NSInteger)quantity
+                         URL:(NSURL *)URL
+                  categories:(nullable NSArray<NSArray<NSString *> *> *)categories
```
Comment (eb-smith-affirm) [inline AffirmSDK/AffirmItem.h]: I'm coming into this cold and don't have context. But an array of arrays is kind of weird to wrap one's head around.

Would it be clearer to define an `AffirmCategory` like:

```obj-c
@interface AffirmCategory
@property (nonatomic, nonnull) NSString name;
@property (nonatomic, nullable) NSArray<NSString*> subCategories;
... etc.
```

and have `categories` be an `NSArray<AffirmCategory*>*`?

**Review by eb-smith-affirm — COMMENTED** [review]
One suggestion. Thanks for adding HelpDoc.

**Comment by eb-smith-affirm** [comment]
Also the tests fail. What do you use as the CI/CD?

File: AffirmSDK/AffirmItem.h (hunk 1 above)
Comment (hoyelo) [reply AffirmSDK/AffirmItem.h]: Right now, both the backend and Android SDK use a two-dimensional string array without any subdivisions. Do we need to standardize this? If we decide to define an AffirmCategory, it only needs to include an array, the name isn't really necessary.

Backend:
https://docs.affirm.com/developers/reference/the-item-object

Android SDK:
https://github.com/Affirm/affirm-merchant-sdk-android/pull/137/files#diff-0f9e2c64c69d37cfd2646df60aecfb5bf5ce6eef5cbee13c2981134d38b47a0eR54

Comment (hoyelo) [reply AffirmSDK/AffirmItem.h]: Due to some issues with GitHub Actions on the latest macOS, downgrading to macOS-12 can resolve the problem. You can find more details here: https://github.com/actions/runner-images/issues/9591.

Comment (bf-affirm) [reply AffirmSDK/AffirmItem.h]: Ed's comment seems fine to me, might help with clarity.
Will we run into any issues in the future with this macOS "downgrade"?

Comment (hoyelo) [reply AffirmSDK/AffirmItem.h]: Sure, I'll optimize it. The downgrade does not impact the SDK, it simply enables tests to run normally.

Comment (hoyelo) [reply AffirmSDK/AffirmItem.h]: @eb-smith-affirm @bf-affirm It has been optimized, thanks for your response.

**Review by eb-smith-affirm — APPROVED** [review]
👍

**Pull request merged**